### PR TITLE
chore: python pinned and cron expression updated

### PR DESCRIPTION
**Pull request by Hamza442**
**Description:** Describe in a couple of sentences what this PR adds

**JIRA:** Link to JIRA ticket

**Dependencies:** dependencies on other outstanding PRs, issues, etc.

**Merge deadline:** List merge deadline (if any)

**Installation instructions:** List any non-trivial installation
instructions.

**Testing instructions:**

1. Open page A
2. Do thing B
3. Expect C to happen
4. If D happened instead - check failed.

**Reviewers:**
- [ ] tag reviewer
- [ ] tag reviewer

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant
solutions, hacks, quick-and-dirty implementations, concerns about
migrations, etc.

### CODE REVIEW

File: .github/workflows/upgrade-python-requirements.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    inputs:
+      branch:
+        description: "Target branch to create requirements PR against"
+        required: true
+        default: '$default-branch'
+
+jobs:
+
+  call-upgrade-python-requirements-workflow:
+     uses: edx/.github/.github/workflows/upgrade-python-requirements.yml@master
```
Comment (jazibhumayun) [inline .github/workflows/upgrade-python-requirements.yml:17]: nit: Please add a comment here to mention which script it invoke.

**Comment by jazibhumayun** [comment]
Looks good :+1:

**Comment by jazibhumayun** [comment]
Please rebase from main branch because I had added one pinned dependency in constraints file recently.

**Review by jazibhumayun — APPROVED** [review]

File: .github/workflows/upgrade-python-requirements.yml (hunk 1 above)
Comment (Hamza442) [reply .github/workflows/upgrade-python-requirements.yml:17]: The Job **call-upgrade-python-requirements-workflow** is invoke the following script **upgrade-python-requirements.yml**. Please follow the attached link to review the script https://github.com/edx/.github/blob/master/.github/workflows/upgrade-python-requirements.yml#L62-L65

File: .github/workflows/upgrade-python-requirements.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+on:
+  schedule:
+    # will start the job at 13:15 UTC on 1st of every month
+    - cron: "15 13 1 * *"
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: "Target branch to create requirements PR against"
+        required: true
+        default: '$default-branch'
```
Comment (estute) [inline .github/workflows/upgrade-python-requirements.yml]: How is $default-branch available? Is it a GH-specific environment variable or do we need to declare it? In the case of this repo, it is `main`, not `master`.

**Review by estute — APPROVED** [review]
Looks good to me. One minor question, but not a blocker.

File: .github/workflows/upgrade-python-requirements.yml (hunk 2 above)
Comment (Hamza442) [reply .github/workflows/upgrade-python-requirements.yml]: I think you are right in this case default should be main

File: .github/workflows/upgrade-python-requirements.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        default: 'main'
+
+jobs:
+
+  call-upgrade-python-requirements-workflow:
+     uses: edx/.github/.github/workflows/upgrade-python-requirements.yml@master
+     with:
+       branch: ${{ github.event.inputs.branch }}
+       team_reviewers: 'edx-data-engineering'
+       email_address: 'analytics@edx.opsgenie.net'
```
Comment (jazibhumayun) [inline .github/workflows/upgrade-python-requirements.yml]: I think we can use `data-engineering@edx.org` instead of `analytics@edx.opsgenie.net`

Comment (Hamza442) [reply .github/workflows/upgrade-python-requirements.yml]: Thanks for letting me know

**Pull request merged**